bpo-46422: use `dis.Positions` in `dis.Instruction` (#30716)
python/cpython: Doc/library/dis.rst

      ``True`` if other code jumps to here, otherwise ``False``

   .. data:: positions

>       Optional namedtuple ``(lineno, end_lineno, col_offset, end_col_offset)``

Reviewer: I think we should explicitly mention the `Positions` class.

Author: Yes, I was thinking about it. So, here are my thoughts.

- If we mention it, we need to document it on the same page
- This page does not have a clear place for it, only below `Instruction` class under `Python Bytecode Instructions` header. It feels foreign to me
- On the other hand, if we don't mention exact class - we have more freedom in the future: it is not "really public" in CPython's rules
- `Positions` itself is not a real "class", it does not have any methods / interesting props / etc. It is just a structure with only 4 self-explanatory fields
- Documentation can be fixed later at any point if needed

Do you agree? 🙂

Reviewer: `Optional namedtuple` does not sound really convincing, my 2cts is that we should document it. You can add the documentation right after the `Bytecode`, since it is practically the only use case.

Author: Done!

      ``True`` if other code jumps to here, otherwise ``False``

   .. data:: positions

>       Optional ::`dis.Positions` namedtuple holding
>       the span of source code covered by this instruction.

Reviewer: We always populate the `positions` with a `Positions` instance.
```suggestion
      :class:`dis.Positions` object holding the
      start and end locations that are covered by this instruction.
 ```

   .. versionadded:: 3.4

   .. versionchanged:: 3.11

      Field ``positions`` is added.

.. class:: Positions

>    Namedtuple with :class:`dis.Instruction` position information.
>    All fields might be ``None`` if some information is missing.

Reviewer: ```suggestion
   In case the information is not available, some fields might be `None`.
```
